src/Chapter3.hs

```diff
@@ -561,21 +665,51 @@ introducing extra newtypes.
     implementation of the "hitPlayer" function at all!
 -}
 data Player = Player
-    { playerHealth    :: Int
-    , playerArmor     :: Int
-    , playerAttack    :: Int
-    , playerDexterity :: Int
-    , playerStrength  :: Int
-    }
+    { playerHealth    :: Health
+    , playerArmor     :: Armor
+    , playerAttack    :: Attack
+    , playerDexterity :: Dexterity
+    , playerStrength  :: Strength
+    } deriving (Show)
+
+newtype Attack = Attack 
+    {
+        attackValue :: Int
+    } deriving (Show)
+newtype Strength = Strength
+    {
+        strengthValue :: Int
+    } deriving (Show)
+newtype Damage = Damage 
+    {
+        damageValue :: Int
+    } deriving (Show)
+newtype Armor = Armor 
+    {
+        armorValue :: Int
+    } deriving (Show)
+newtype Dexterity = Dexterity 
+    {
+        dexterityValue :: Int
+    } deriving (Show)
+newtype Defense = Defense 
+    {
+        defenseValue :: Int
+    } deriving (Show)
+newtype Health = Health 
+    {
+        healthValue :: Int
+    } deriving (Show)
 
-calculatePlayerDamage :: Int -> Int -> Int
-calculatePlayerDamage attack strength = attack + strength
 
-calculatePlayerDefense :: Int -> Int -> Int
-calculatePlayerDefense armor dexterity = armor * dexterity
+calculatePlayerDamage :: Attack -> Strength -> Damage
+calculatePlayerDamage attack strength = Damage (attackValue attack + strengthValue strength)
 
-calculatePlayerHit :: Int -> Int -> Int -> Int
-calculatePlayerHit damage defense health = health + defense - damage
+calculatePlayerDefense :: Armor -> Dexterity -> Defense
+calculatePlayerDefense armor dexterity = Defense (armorValue armor * dexterityValue dexterity)
+
+calculatePlayerHit :: Damage -> Defense -> Health -> Health
+calculatePlayerHit damage defense health = Health (healthValue health + defenseValue defense - damageValue damage)
```

Comment:
Nice work on newtypes! :medal_sports: Newtypes are one very important feature in Haskell :+1:
You can read more about the newtype pattern in our blog post: https://kowainik.github.io/posts/haskell-mini-patterns#newtype